backend/app/usecase/url/urlcreator.go

```diff
@@ -73,6 +80,9 @@ func (c CreatorPersist) CreateURL(url entity.ShortLink, customAlias *string, use
 	}
 
 	if !c.aliasValidator.IsValid(customAlias) {
+		if c.aliasValidator.HasFragmentCharacter(*customAlias) {
+			return entity.ShortLink{}, ErrAliasWithFragment(*customAlias)
+		}
```

Comment:
I think we should modify `IsValid` interface to return `boolean, Violation`. Violation is an enum. We are not 100% sure it's fragment violation just by calling this function here.

Reply:
Then we can modify `ErrInvalidCustomAlias` include a field call `violation`

Reply:
`ErrInvalidCustomAlias` has to be changed to a struct instead.

Reply:
Done